packages/repack/src/modules/ScriptManager/types.ts

```diff
@@ -88,6 +88,28 @@ export interface ScriptLocator {
    * Setting this flat to `false`, disables that behavior.
    */
   cache?: boolean;
+  /**
+   * Function called before loading or getting from the cache and after resolving the script locator.
+   * It's an async function which should return a boolean indicating whether the script should be loaded or use default behaviour.
+   * This is useful when you want to load a script only when certain conditions are met
+   * (e.g. ask user if they want to update/download new version of the script)
+   *
+   * When `true` is returned, the script will be loaded from the network.
+   * When `false` is returned, the script will be loaded from the cache.
+   *
+   * @param scriptId Id of the script to resolve.
+   * @param caller Name of the calling script - it can be for example: name of the bundle, chunk or container.
+   * @param isScriptCacheOutdated Boolean indicating whether the script cache is outdated or not. It's `true` when the script
+   * cache is outdated and `false` when the script cache is up to date or there is no cache for the script.
+   * Outdated cache means that the script was previously downloaded and put into cache,
+   * but the script locator data (method, url, query, headers, or body) has changed since then.
+   * @returns Boolean indicating whether the script should be loaded or not
+   */
+  shouldUpdateScript?: (
```

Comment:
here I also think might be better to rename to e.g. `getShouldUpdateScript` - the goal here is to point within it's name it's a method